### PR TITLE
Add `CITATION.cff`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Context:

https://github.blog/2021-08-19-enhanced-support-citations-github/
https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-citation-files

Spec:
https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md

See the result in:
https://github.com/PyTorchLightning/pytorch-lightning/tree/github/cff

to the right, below "About" there is now a button "cite this repository"

image

APA:
```apa
William, F., & The PyTorch Lightning team. (2019). PyTorch Lightning (Version 1.4.4) [Computer software]. https://doi.org/10.5281/zenodo.3828935`
```

BibTeX
```bibtex
@misc{William_PyTorch_Lightning_2019,
author = {William, Falcon and {The PyTorch Lightning team}},
doi = {10.5281/zenodo.3828935},
month = {3},
title = {{PyTorch Lightning}},
url = {https://github.com/PyTorchLightning/pytorch-lightning},
year = {2019}
}
```

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,18 @@
+cff-version: 1.2.0
+message: "If you want to cite the framework, feel free to use this (but only if you loved it 😊)"
+title: "PyTorch Lightning"
+abstract: "The lightweight PyTorch wrapper for high-performance AI research. Scale your models, not the boilerplate."
+date-released: 2019-03-30
+authors:
+  - family-names: "William"
+    given-names: "Falcon"
+  - name: "The PyTorch Lightning team"
+version: 1.4.4
```
[the remark this answers is not in the document]
Comment (Borda) [reply CITATION.cff:10]: can we lower the granularity?
```suggestion
version: 1.4
```

**Review by Borda — APPROVED** [review]
let's enable the zenodo first

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+doi: 10.5281/zenodo.3828935
```
[the remark this answers is not in the document]
Comment (carmocca) [reply CITATION.cff:11]: @Borda feel free to fix the doi once zenodo is enabled and set auto-merge :)

[the remark this answers is not in the document]
Comment (Borda) [reply CITATION.cff:11]: seems the connection for the automatic update is not active yet... @lexierule

[the remark this answers is not in the document]
Comment (carmocca) [reply CITATION.cff:11]: Merging, the doi can be updated later

**Pull request merged**